frontend/src/i18n/request.ts

```diff
@@ -0,0 +1,62 @@
+import merge from "lodash/merge";
+import { getRequestConfig } from "next-intl/server";
+import { cookies, headers } from "next/headers";
+import { defaultLocale, supportedLocales } from "./locales";
+import enLocale from "../../locales/en.json";
+
+export default getRequestConfig(async () => {
+  const headersList = await headers();
+
+  let currentUserLocale = (await cookies())?.get("USER_LOCALE")?.value;
+  if (currentUserLocale && !supportedLocales.includes(currentUserLocale)) {
+    currentUserLocale = undefined;
+  }
+  const locale =
+    currentUserLocale ||
+    getLocaleFromHeaders(headersList, supportedLocales) ||
```

Comment:
Remove this function and let the `defaultLocale` be the one applied by default.

Reply:
Fixed 😁